### DISCUSSION TITLE
Member utilisation

### DISCUSSION THREAD

**Discussion by glennawatson** [dotnet-foundation/Home#52] [General Discussions]
@perksey is attributed for the following

I do think that we’ve missed a trick in that the actual community members who are already passionate about the platform the foundation is trying to push is the foundation’s most valuable asset but remains mostly unwielded by the foundation in their efforts.

Resources necessary to promote .NET need not be director exclusive. Focus on a passionate member and active community. 

Delegate more project support responsibilities from directors onto the project committee to increase response time. Project committees are pushing issues straight back to directors at the moment.

**Comment by glennawatson** [comment]
Also, how do you go about incentivising people to contribute as a project committee? 

Eg probably not enough people at the moment and there's nothing to entice people to do the DNF grunt work at the moment.

**Reply by rprouse** [reply]
I think this gets to the root of any OSS project, how do we get people to volunteer their time to help? If anyone can do it, it is our community that has spent their careers building community around their projects.

I don't have the answers but I was thinking about this last night. I'd like to encourage more junior members of the community to join the committees. I think we need to work on letting them know they are needed and that their skills are valuable to us. I think we also need to show them the benefits. Personally I think it looks great on your resume and it is a chance to meet and work with other people from the community.

Can anyone else add to or define the benefits better than me? I'm better with code than words 😀

**Reply by Perksey** [reply]
> I'd like to encourage more junior members of the community to join the committees

We definitely do need more members involved in the committees yeah. Ideally, even passive members have resources that they can utilise if they feel like it to help promote .NET and the ecosystem therein, but that's a mammoth problem that will take lots of time to solve and something that can be solved if the committees are empowered to do so :)

But also, this is very much pot calling the kettle black as I am not involved in any committees today.

**Comment by Perksey** [comment]
> Delegate more project support responsibilities from directors onto the project committee to increase response time. Project committees are pushing issues straight back to directors at the moment.

This is something that I have spoken to @tannergooding (of the Projects Committee) about informally, but one thing that has always stuck out is how project support responsibility has always fallen on one or two people (previously Claire, occasionally with the help of Chris) whereas we should probably have a pool of volunteers for this to prevent some individuals getting overworked as well as increasing support turnaround.

**Reply by rprouse** [reply]
> one thing that has always stuck out is how project support responsibility has always fallen on one or two people (previously Claire, occasionally with the help of Chris) whereas we should probably have a pool of volunteers for this to prevent some individuals getting overworked as well as increasing support turnaround.

I think that this is key. I think that a lot of the mistakes that have been made were because too few people were trying to do too many things and corners were cut.

**Reply by Perksey** [reply]
Admittedly I have thought about getting involved in the projects committee, but being the go/no-go poll for project onboarding it is today I didn't really feel the committee needed an extra hand. From my perspective, I'd love to help out with support the diverse roster of projects in the foundation and help free up the time of the directors who have enough on their plate as is as well as ensure projects get swift responses!

But for this, the board will need to ensure they have delegated the necessary access and rights to the committee to facilitate this.

**Reply by tannergooding** [reply]
-- Just noting to start, the volunteer work I've done in the foundation and with the project committee is a personal investment and not related to my job on the .NET Libraries team. As such, my comments/views/opinions here are my own.

I think, in particular there is a lot detailed under https://github.com/dotnet-foundation/projects, but of those the monthly meetings tend to only cover the first responsibility "Review and evaluate new Project applications".

There is a lot of opportunity to do more; but as indicated above this depends on people volunteering their time to make it happen. This generally comes about, as much of OSS does, from people in the community that are particularly passionate about the thing in question. Likewise, I think there is a lot that the foundation can do to better the ecosystem even without projects being direct members.

-----------------

Some of the ideas that I've had and/or raised in the past include:
* Having a path for the community to recommend and establish new responsibilities, features, or asks for a committee to take on or that the foundation should provide
  * In my mind these recommendations do not necessarily have to be done by the committee, but rather driven by the committee/foundation overall the ask does happen and provide the overall guidance on ensuring it meets expectations -or- to provide a clear response as to why a direction can't or won't be taken. The actual execution/completion of the action can entirely happen based on volunteers from the foundation who are interested.
* Providing additional visibility to projects; whether they are member projects; new ideas that people in the community are starting; or even long-standing non-member projects that are looking for feedback, contributions, or other resources
* Providing clear documentation on what the foundation provides and how a project can achieve those same provisions independently of the foundation
  * Many things such as CLA bots, documentation sites, source browsers, and even code signing are possible independently of the foundation. Projects should be able to succeed even if they never plan on having direct membership simply by utilizing the same or similar resources to that of the .NET Foundation
* Providing documentation covering "best practices" and why the practices exist
  * There are many things that can help make a project successful but no one "true" path. Having documentation (or even templates) pointing users to resources and reasons why certain patterns are good can help guide them to making choices that help set themselves up for success. Many of these items are simply following a particular pattern or utilizing a free resource and can be independent of a project being in the foundation in the first place.
  * Almost all of these practices are optional, so projects should be able to read a given item and decide if it is right for them or not
  * Some of these include things like choosing the right license, setting up a CLA bot, deciding on a code of conduct, setting up CI testing, what should be covered in your readme/contributing docs, what needs to go into your `Third Party Notices`, setting up a source browser, setting up a documentation website, centralizing your project options/targets, picking a testing framework, knowing when its right to target .NET Framework vs .NET Standard vs just .NET, setting up build scripts vs ensuring `dotnet build` just works, etc
  * Many of these things are also opinionated or evolve over time; so this documentation can/should be extensible and influenced by the community
* And possibly the hardest... Find a way to help large projects continue their success and maintain themselves
  * Starting an OSS project tends to be simple; particularly when you are the only person working on it. As time evolves, you make your first releases, and people begin depending on the project things start getting more complex.
  * One aspect is that many OSS projects are not backed by big enterprises; they are run by "volunteers" in their spare time. These maintainers typically have full time jobs of their own to help support themselves and so the balance of supporting themselves vs maintaining the project is constantly in the balance.
  * Another aspect is that many popular open source projects are filling a particular scenario and it may require expertise that makes getting contributions hard; even when there are people who might be otherwise willing
  * The easiest thing to say is "just pay the maintainers of major projects or get them enough regular donations so they can work on it full time", but that's complex in practice for many reasons. Finding additional maintainers/volunteers can be just as complex/difficult, especially when it comes to ensuring the project maintains the intended direction of the primary owner. However, there is likely some solution that the community and foundation can come to that helps fit the needs and real world limitations that exist here.

**Comment by mitchelsellers** [comment]
Fundamentally I do not believe that there is even a committee or group formed for this action. Defining what and how people can get involved is the number one driver to letting people in. 

The current project committee is solely documented as “onboarding decisions”. Not long term project support.   

Some of the other related issues, which as what benefits do you really get them compound into this discussion as without clear notes on that supporting ongoing needs are hard. 

So possibly expanding the current committees purpose. Or creating a new one could help channel some volunteerism. But if doing so actions must be delegated so the group can thrive on its own.

**Reply by rprouse** [reply]
There are already several committees formed all of which need help. We are also forming the new maintainers committee. There is some overlap in that with the current Project committee, so @devlead and I need to talk with @sbwalker to define the boundaries. He has a lot more experience than we do, so any help he can give will be appreciated.

If anyone wants to join or help our existing committees, I'd recommend reaching out to their chairs or other people on the committee. Take a look at https://dotnetfoundation.org/community/committees and see what you are passionate about.

**Reply by sbwalker** [reply]
@rprouse the next scheduled Project Committee meeting is Oct 18 at 12 PM ET and I would be happy to add you to the invite. I am getting married today so I have personally not had any time to focus on anything else this past week ;)

**Reply by dansiegel** [reply]
@sbwalker for crying out loud what are you doing on here... go get married, enjoy your special day and CONGRATULATIONS!

**Reply by Perksey** [reply]
So it sounds like the Projects and oddly enough Outreach committees are the ones that strike my eye - I would've thought membership, but their scope seems to be quite slim...

I'm not really used to getting involved in stuff like this, but certain people are trying to shake me out of that which I can do nothing but thank them for ;)

cc @sbwalker (but not before there's a ring on your finger) @shawnwildermuth - would love to get involved in those two committees. Email is dylan dot perks at l3harris dot com.
